pkg/forwarder/forwarder_health.go

```diff
@@ -111,17 +110,18 @@ func (fh *forwarderHealth) healthCheckLoop() {
 	}
 }
 
-// computeDomainsURL populates a map containing API Endpoints per API keys that belongs to the forwarderHealth struct
-func (fh *forwarderHealth) computeDomainsURL() {
-	for domain, apiKeys := range fh.keysPerDomains {
+// computeKeysPerAPIDomain populates a map containing API Endpoints per API keys that belongs to the forwarderHealth struct
+func (fh *forwarderHealth) computeKeysPerAPIDomain(keysPerDomains map[string][]string) {
```

Comment:
actually, there's already some logic in the `config` package to make changes to the configured domains depending on their values, see https://github.com/DataDog/datadog-agent/blob/8890126df2160940d9a754a57f52a756162ae957/pkg/config/config.go#L884-L900

For consistency I think it would make sense to move the logic that takes a `domain` and returns the related `APIDomain` to a function in the config package (for example `func ComputeAPIDomain(domain string) string`).